scalafmt-cli-sbt/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala

```diff
@@ -0,0 +1,25 @@
+package org.scalafmt.sbt
+import scala.meta.internal.tokenizers.PlatformTokenizerCache
+import sbt._
+import Keys._
+
+object ScalafmtPlugin extends AutoPlugin {
+  override def trigger: PluginTrigger = allRequirements
+  object autoImport {
+    val scalafmt: Command =
+      Command.args("scalafmt", "run the scalafmt command line interface.") {
+        case (state, args) =>
+          org.scalafmt.cli.Cli.main("--non-interactive" +: args.toArray)
+          PlatformTokenizerCache.megaCache.clear()
```

Comment:
You want to run this in the new plugin too, megaCache will eventually go away but for now it's best to call it on every file. https://github.com/scalameta/scalameta/issues/1068

Reply:
Call on every file? Your plugin calls it once at start.

Reply:
And when I should call it? Even when I check source or only when format?

Reply:
It's a bit trickier to decide compared to the cli, but I think we can do it once per config/project. This should ideally run whenever scalafmt is not running, otherwise it shoudl be a well-behaving cache and the only penalty for clearing it too often is a very minor slowdown hit.